Tests/SubprocessTests/IntegrationTests.swift

```diff
@@ -310,6 +310,82 @@ extension SubprocessIntegrationTests {
         )
     }
 
+    @Test func testEnvironmentInheritOverrideUnsetValue() async throws {
+        #if os(Windows)
+        // First insert our dummy environment variable
+        "SubprocessTest".withCString(encodedAs: UTF16.self) { keyW in
+            "value".withCString(encodedAs: UTF16.self) { valueW in
+                SetEnvironmentVariableW(keyW, valueW);
```

Comment:
nit: some stray semicolons here.

I could also see the set-defer-unset pattern being problematic for concurrently running tests. We had the same problem in Swift Build and adopted a mock-based approach with Swift Concurrency task locals, which works pretty nicely. Not suggesting it's something that needs to be fixed now, but just a thought for future enhancements.